src/utility.js

```diff
@@ -511,3 +511,122 @@ module.exports.proxyConfiguration = async ({
+                    finished = true;
+                    return;
+                }
+            } else {
+                resourcesStats.matchNumber = 0;
+                resourcesStats.oldRequested = resourcesStats.newRequested;
+            }
+            // check if timeout has been reached
+            if (timeoutSecs !== 0 && (Date.now() - startTime) / 1000 > timeoutSecs) {
+                clearInterval(checkFinished);
+                finished = true;
+            }
+        }, CHECK_INTERVAL_MILLIS);
+        const doScroll = async () => {
+            /* istanbul ignore next */
+            await page.evaluate(async (scrollHeightIfZero) => {
+                const delta = document.body.scrollHeight === 0 ? scrollHeightIfZero : document.body.scrollHeight;
+                window.scrollBy(0, delta);
+            }, SCROLL_HEIGHT_IF_ZERO);
+        };
+        const maybeClickButton = async () => {
+            const button = await page.$(buttonSelector);
+            // Box model returns null if the button is not visible
+            if (button && await button.boxModel()) {
+                await button.click({delay: 10});
+            }
+        };
+        while (!finished) {
+            await doScroll();
+            await page.waitForTimeout(50);
+            if (scrollDownAndUp) {
+                await page.evaluate(() => {
+                    window.scrollBy(0, -1000);
+                });
+            }
+            if (buttonSelector) {
+                await maybeClickButton();
+            }
+            if (stopScrollCallback) {
+                if (await stopScrollCallback()) {
+                    break;
+                }
+            }
+        }
+    };
+
+    const commentSelector = 'ytd-comment-thread-renderer';
+    // Scroll first to load at lease one comment.
+    await page.evaluate(() => {
+        window.scrollBy(0, 500);
+    });
+    await page.waitForSelector('ytd-comment-thread-renderer');
+    await infiniteScroll(page, {stopScrollCallback:async () => {
+            const commentCount = await page.evaluate(()=>{
+                return document.body.querySelectorAll(commentSelector).length;
+            });
+            log.debug(`Got ${commentCount}/${maxCommentCount} comments for ${page.url()}`)
+            return commentCount >= maxCommentCount && maxCommentCount > 0;
+        }}
+    );
+    const comments = await page.evaluate((max) => {
+        const elements = document.body.querySelectorAll(commentSelector);
+        const a = [];
+        for (let i =0; i < elements.length; i++) {
+            const e = elements[i];
+            const author = e.querySelector('#author-text > span').innerHTML.trim()
+                .replace(/\\n/g, '');
+            if (author) {
+                const comment = e.querySelector('#content-text').innerHTML.trim()
+                    .replace(/\\n/g, '');
+                a.push({
+                    author: author,
+                    comment: comment,
```

Comment:
I would probably call it `text` or something else than `comment` because we call the top level object like that